### PR TITLE
Add forms completed

### PR DESCRIPTION
**Pull request by ometman**
I have completed the forms on branch 'add-forms':

- Created a method called current_user in ApplicationController to make current user data available to all controllers.
   - The current_user method returns the first user from the database.
   - Proper authentication was not implemented here.
- 2 Forms were created to perform the following functions:
   - Create a Post on behalf of the current_user
   - Create a comment on behalf of the current_user
- Users were allowed to add likes to Posts.

### CODE REVIEW

File: db/migrate/20240117035357_remove_text_from_likes.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class RemoveTextFromLikes < ActiveRecord::Migration[7.1]
+  def change
+    remove_column :likes, :text
+  end
+end
```
Comment (danmainah) [inline db/migrate/20240117035357_remove_text_from_likes.rb]: - [ ] @ometman Kindly consider removing this migration file as you don't have a column text in the likes table, its generating a migration error.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+
+    if @post.save
+      flash[:success] = 'Post saved successfully'
+      redirect_to user_posts_path(@user)
+    else
+      flash[:error] = 'Error: Post could not be saved'
+      flash[:error_details] = @post.errors.full_messages.join(', ')
+      render :new
+    end
   end
```
Comment (danmainah) [inline app/controllers/posts_controller.rb:28]: - [ ] @ometman Great work adding a method to add new posts, however, we are supposed to use the `current_user` method created on the application controller to create a new post on behalf of the current user, kindly consider using the method to get your user.
image

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    puts "Post: #{@post.inspect}"
+    puts "Comment: #{@comment.inspect}"
+
+    if @comment.save
+      flash[:success] = 'Comment was successfully created.'
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new
+    end
+  end
```
Comment (danmainah) [inline app/controllers/comments_controller.rb]: - [ ] @ometman Great work adding a method to add new comments, however, we are supposed to use the current_user method created on the application controller to create a new comment on behalf of the current user, kindly consider using the method when creating the comment.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @ometman , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added the current_user method in the ApplicationController to make current user data available to all controllers.
✔️  Added action methods to create posts, comments, and likes
✔️  Added forms to add new posts, comments and a button to add likes to a post.



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: db/migrate/20240117035357_remove_text_from_likes.rb (hunk 1 above)
Comment (ometman) [reply db/migrate/20240117035357_remove_text_from_likes.rb]: Added a column text and used that to take it out, but I guess later too it out manually. 😄

**Review by julie-ify — APPROVED** [review]
Hi @ometman,

## Highlights!
- Good code organization and working forms ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (ometman) [reply app/controllers/posts_controller.rb:28]: Fixed. thanks

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (ometman) [reply app/controllers/comments_controller.rb]: Fixed. Thanks

**Comment by ometman** [comment]
Thanks

**Pull request merged**